### DISCUSSION TITLE
Lucid Model: Is there a way to fetch all ids from a given many-to-many relationships?

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#1331] [Help] [answered]
Hello, 🙂 

I have two models `List` and `Company` that are linked together with a `many-to-many` relationship.

When I fetch a list I would like to add in the object the property `company_ids: [1, 3, 29, 288]` (it includes all the ids of the companies that belong to the list).

```
const Model = use('Model')

class List extends Model {
  companies() {
    return this.belongsToMany('App/Models/Company')
  }

  company_ids() {
    return this.companies().fetch().ids()
  }
}

module.exports = List
```

This code does not work when doing:

```
const lists = await List.query().where('team_id', team.id).with('company_ids').fetch()
```

I know I am doing something wrong but is there an easy way to achieve this goal?

**Comment by hkmsadek** [comment, marked as answer]
Why don't you use the relation itself? 

May be try like this 

```
companies() {
    return this.belongsToMany('App/Models/Company').select('id')
  }

```

**Reply by LeCoupa** [reply]
Thank you! 😃
